libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/lifecycle/UIComponent.kt

```diff
@@ -35,3 +35,32 @@ open class UIComponent : MapboxNavigationObserver {
         coroutineScope.launch { collect(action) }
     }
 }
+
+/**
+ * Return UIComponent that gets re-created using [factory] when [value] changes.
+ *
+ * eg.
+ * ```
+ *   reloadOnChange(myFlowValue) { value ->
+ *      MyComponent(value)
+ *   }
+ * ```
+ */
+fun <T> reloadOnChange(value: Flow<T>, factory: (value: T) -> UIComponent): UIComponent =
+    object : UIComponent() {
+        private var childComponent: UIComponent? = null
+
+        override fun onAttached(mapboxNavigation: MapboxNavigation) {
+            super.onAttached(mapboxNavigation)
+            value.observe {
+                childComponent?.onDetached(mapboxNavigation)
+                childComponent = factory(it)
+                childComponent?.onAttached(mapboxNavigation)
+            }
+        }
+
+        override fun onDetached(mapboxNavigation: MapboxNavigation) {
+            super.onDetached(mapboxNavigation)
+            childComponent?.onDetached(mapboxNavigation)
+        }
+    }
```

Comment:
This is a great utility. With route line options and customizations I don't see any issue when detaching and reattaching every time the option is changed. However, I am not sure of any implications for other complex UI views that would detach and reattach based on options. I guess we will learn that in time and improve if needed. But so far this looks good to me.

Reply:
@tomaszrybakiewicz I have approved the PR, but it would be great to add a test for this if possible